### DISCUSSION TITLE
How to reshape "100x1e" irreps for force prediction

### DISCUSSION THREAD

**Discussion by TommyDzh** [e3nn/e3nn#442] [Q&A] [answered]
Let's say I have **h**= o3.Irreps("100x1e").randn(N, -1), the tensor shape is (N, 100*3). Now I want to use a Linear layer to predict forces, e.g. NN = nn.Linear(100,1). My problem is how should I reshape **h** into **h'** and send **h'** \in [N, 3, 100] into F = NN(**h'**)?
1. **h'** = **h**.reshape(N, 3, 100)
2. **h'** = **h**.reshape(N, 100, 3).permute(0,2,1)

**Comment by mariogeiger** [comment, marked as answer]
2

**Reply by TommyDzh** [reply]
Thanks!
